libcodechecker/analyze/analysis_manager.py

```diff
@@ -366,15 +373,39 @@ def check(check_data):
                         LOG.debug(str(ex))
                         other_files = set()
 
+                    mentioned_files_dependent_files = set()
+                    for of in other_files:
+                        mentioned_file = os.path.abspath(
+                            os.path.join(action.directory, of))
+                        # Use the target of the original build action
+                        key = mentioned_file, action.target
+                        mentioned_file_action = actions_map.get(key)
+                        if mentioned_file_action is not None:
+                            mentioned_file_deps = get_dependent_headers(
+                                mentioned_file_action, archive)
+                            mentioned_files_dependent_files.\
+                                update(mentioned_file_deps)
+                        else:
+                            LOG.debug("Could not find %s in build actions"
+                                      % key)
+
                     dependencies.update(other_files)
+                    dependencies.update(mentioned_files_dependent_files)
 
-                    LOG.debug("Writing dependent files to archive.")
+                    # `dependencies` might contain absolute and relative paths
+                    # for the same file, so make everything absolute.
+                    dependencies_copy = set()
                     for dependent_source in dependencies:
                         dependent_source = os.path.join(action.directory,
```

Comment:
I know this is unrelated to this patch but this does not seem to be correct for me at all.
If `dependent_source` is absolute we should not join it to `action.directory`. If it is relative, the result of join must be absolute, since `action.directory` must be absolute. So the next check is redundant.

Reply:
Yes, right it will be always absolute, so the check is redundant, i am removing it.

Other than that, if the `dependent_source` is absolute than it remains absolute after the join, that is how `os.path.join` behaves and documented:
> If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.